src/Gui/QuantitySpinBox.cpp

```diff
@@ -709,7 +709,7 @@ void QuantitySpinBox::clearSchema()
     updateText(d->quantity);
 }
 
-QString QuantitySpinBox::getUserString(const Base::Quantity& val, double& factor, QString& unitString) const
```

Comment:
Qt uses QString everywhere in its widget classes. I think  we keep it like this. Or is there reason that requires this change?

Reply:
The whole idea is to keep `QString` everywhere in interacts with Qt widgets. I'd like to see `std::string` to be used in the core, but GUI is written in Qt, so it should stick with `QString`. There still might me inconsistencies as I was trying different approaches. Will recheck that.

Reply:
This particular one should stay `QString`, change reverted.

Reply:
Using a quantity spin box with German locale causes a serious regression. When I create a Part box the default values are 10 mm and they used to be displayed as `10,00000 mm` but now the comma is replaced with a dot which makes the values bigger by a factor of 100 000

Reply:
The bug must be caused by transforming the quantity into a locale-specific string, i.e. in the units scheme classes because in other parts of the program (e.g. the dimensions of the viewport size) it also uses the wrong decimal separator.

Reply:
Yes, that decimal separator has no easy solution, so I used your proposal above to leave original implementation.